### PR TITLE
Add options for bash scripts

### PR DESCRIPTION
**Pull request by cd1989**
### Ⅰ. Describe what this PR did

Add options to bash scripts:

```
set -o nounset
set -o errexit
```
Otherwise the `build.sh` will continue when some steps failed. The final result is build a image with missing artifacts.

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -1,4 +1,8 @@
 #!/bin/bash
+
+set -o nounset
```
Comment (Starnop) [inline hack/build.sh]: Maybe it's better to use `set -euo pipefail` instead.

**Comment by Starnop** [comment]
In addition, could you please help to make a check for all scripts and do the same thing for them. THX >

File: hack/build.sh (hunk 1 above)
Comment (cd1989) [reply hack/build.sh]: I'd prefer the following one, it's more readable.
```
set -o nounset
set -o errexit
set -o pipefail
```

**Comment by cd1989** [comment]
Sure

**Comment by Starnop** [comment]
LGTM.

**Pull request merged**